File: Data Structures & Algorithms/top-k-elements-in-list/submission-7.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        int[] res = new int[k];
        HashMap<Integer,Integer> count = new HashMap<>();
        List<Integer>[] freq = new List[nums.length + 1];
        for (int i = 0 ; i<freq.length; i++){
            freq[i] = new ArrayList();
        }
        for (int i = 0;i<nums.length;i++){
            count.put(nums[i],1+count.getOrDefault(nums[i],0));
        }
        for (int num : count.keySet()){
            int t = count.get(num);
            freq[t].add(num);
        }
        int index = 0;
        for (int i = freq.length-1;i>0 && index<k;i--){
            for(int n:freq[i]){
                res[index++]=n;
                if (index == k){
                    return res;
                }
            }
        }
        return res;
    
    }
    
}

File: Data Structures & Algorithms/climbing-stairs/submission-7.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 2  or n == 1:
            return n
        res = [0]*(n+1)
        res[1]=1
        res[2]=2
        
        for i in range(3,n+1):
            res[i] = res[i-1]+res[i-2]
        print (res[i])
        return res[i]


        